bots/pr/src/test/java/org/openjdk/skara/bots/pr/IntegrateTests.java

```diff
@@ -1296,4 +1295,103 @@ void retryAfterInterruptExtraChange(TestInfo testInfo) throws IOException {
+                    .addReviewer(reviewer.forge().currentUser().id())
+                    .addAuthor(badIntegrator.forge().currentUser().id())
+                    .addCommitter(integrator.forge().currentUser().id());
+            var mergeBot = PullRequestBot.newBuilder().repo(botUser).censusRepo(censusBuilder.build()).build();
+
+            // Populate the projects repository
+            var localRepo = CheckableRepository.init(tempFolder.path(), author.repositoryType());
+            var masterHash = localRepo.resolve("master").orElseThrow();
+            assertFalse(CheckableRepository.hasBeenEdited(localRepo));
+            localRepo.push(masterHash, author.url(), "master", true);
+
+            // Make a change with a corresponding PR
+            var editHash = CheckableRepository.appendAndCommit(localRepo);
+            localRepo.push(editHash, author.url(), "refs/heads/edit", true);
+            var authorPr = credentials.createPullRequest(author, "master", "edit", "This is a pull request");
+
+            // Approve it as another user
+            var reviewerPr = reviewer.pullRequest(authorPr.id());
+            reviewerPr.addReview(Review.Verdict.APPROVED, "Approved");
+
+            // Issue /integrate defer command and verify the PR gets deferred
+            authorPr.addComment("/integrate defer");
+            TestBotRunner.runPeriodicItems(mergeBot);
+            var deferred = authorPr.comments().stream()
+                    .filter(comment -> comment.body().contains("Integration of this pull request has been deferred"))
+                    .count();
+            assertEquals(1, deferred, "Missing deferred message");
+            assertTrue(authorPr.labelNames().contains("deferred"));
+
+            // Try to integrate by non committer
+            var badIntegratorPr = badIntegrator.pullRequest(authorPr.id());
+            badIntegratorPr.addComment("/integrate");
+            TestBotRunner.runPeriodicItems(mergeBot);
+            var onlyCommitters = authorPr.comments().stream()
+                    .filter(comment -> comment.body()
+                            .contains("Only project committers are allowed to issue the `integrate` command on a deferred pull request."))
+                    .count();
+            assertEquals(1, onlyCommitters, "Missing error about only committers can integrate");
+
+            // Issue /integrate undefer and verify the PR is no longer deferred
+            authorPr.addComment("/integrate undefer");
+            TestBotRunner.runPeriodicItems(mergeBot);
+            var undeferred = authorPr.comments().stream()
+                    .filter(comment -> comment.body().contains("Integration of this pull request is no longer deferred and may only be integrated by the author"))
+                    .count();
+            assertEquals(1, undeferred, "Missing undeferred message");
+            assertFalse(authorPr.labelNames().contains("deferred"));
+
```

Comment:
Is it worth attempting to `/integrate` as the integrator here (which should fail since you've set it back to manual)?